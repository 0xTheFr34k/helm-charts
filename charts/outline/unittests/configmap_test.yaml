suite: test configmap

templates:
  - configmap.yaml

tests:
  - it: should match snapshot of default values
    release:
      name: outline
      namespace: outline
    chart:
      version: 1.0.0
      appVersion: 1.0.0
    asserts:
      - matchSnapshot: { }
