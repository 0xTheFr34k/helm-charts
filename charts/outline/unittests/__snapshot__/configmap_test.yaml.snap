should match snapshot of default values:
  1: |
    apiVersion: v1
    data:
      entrypoint.sh: |
        #!/bin/bash

        # Construct the JSON string with host, port, username, and password from environment variables
        JSON=$(printf '{"host":"%s","port":%d,"username":"%s","password":"%s"}' \
          "$REDIS_HOST" "$REDIS_PORT" "$REDIS_USERNAME" "$REDIS_PASSWORD")

        # Base64-encode the JSON string
        BASE64_JSON=$(echo -n "$JSON" | base64)

        # Set the REDIS_URL environment variable
        export REDIS_URL="ioredis://$BASE64_JSON"

        # Execute the original command (yarn start)
        exec yarn start
    kind: ConfigMap
    metadata:
      labels:
        app: outline
        chart: outline-1.0.0
        heritage: Helm
        release: outline
      name: outline-entrypoint
